base/collect_database.go

```diff
@@ -0,0 +1,299 @@
+package base
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"path/filepath"
+	"strconv"
+	"strings"
+
+	"github.com/itchyny/gojq"
+	log "github.com/iter8-tools/iter8/base/log"
+
+	"text/template"
+	"time"
+
+	"sigs.k8s.io/yaml"
+)
+
+type CollectDatabaseTemplateInput struct {
+	MonitoringEndpoint string `json:"monitoringEndpoint" yaml:"monitoringEndpoint"`
+	IAMToken           string `json:"IAMToken" yaml:"IAMToken"`
+	GUID               string `json:"GUID" yaml:"GUID"`
+}
+
+type CollectDatabaseTemplate struct {
+	Url      string            `json:"url" yaml:"url"`
+	Headers  map[string]string `json:"headers" yaml:"headers"` // TODO: headers can be anything?
+	Provider string            `json:"provider" yaml:"provider"`
+	Method   string            `json:"method" yaml:"method"` // TODO: make enum
+	Metrics  []Metric          `json:"metrics" yaml:"metrics"`
+}
+
+type Metric struct {
+	Name         string   `json:"name" yaml:"name"`
+	Description  string   `json:"description" yaml:"description"`
+	Type         string   `json:"type" yaml:"type"` // TODO: make enum
+	Units        string   `json:"units" yaml:"units"`
+	Params       []Params `json:"params" yaml:"params"`
+	JqExpression string   `json:"jqExpression" yaml:"jqExpression"`
+}
+
+type Params struct {
+	Name  string `json:"name" yaml:"name"`
+	Value string `json:"value" yaml:"value"`
+}
+
+// collectDatabaseInputs holds all the inputs for this task
+//
+// Inputs for the template:
```

Comment:
I'm not really sure where I should document all of these inputs so I will put it here for now.

Reply:
Yes, it is CE specific so it probably should not be here. It is documented in the CE template as well but I feel its usage is not clear enough and additional documentation is needed.